src/OrbitBase/include/OrbitBase/NotFoundOr.h

```diff
@@ -20,44 +20,64 @@ struct NotFound {
 };
 
 // NotFoundOr can be used as the return type of a search operation, where the search can be
-// unsuccessful and returns a message. Based on std::variant. Check whether a NotFoundOr object is
+// unsuccessful and returns a message. Check whether a NotFoundOr object is
 // a "not found" result with orbit_base::IsNotFound. Get the message with GetNotFoundMessage.
+//
+// `NotFoundOr<T>` is implemented by privately deriving from `std::variant`. This allows us to take
+// advantage of std::variant's carefully crafted constructors while still keeping std::variant an
+// implementation detail. Implementing those constructors by hand would add a lot of complexity.
 template <typename T>
-using NotFoundOr = std::variant<VoidToMonostate_t<T>, NotFound>;
+class NotFoundOr : private std::variant<VoidToMonostate_t<T>, NotFound> {
+ public:
+  using Value = VoidToMonostate_t<T>;
 
-// Free function to check whether a NotFoundOr type is "not found". Abstracts
-// std::holds_alternative
+  [[nodiscard]] bool HasValue() const { return this->index() == 0; }
+  [[nodiscard]] bool IsNotFound() const { return !HasValue(); }
+
+  [[nodiscard]] const Value& GetValue() const& { return std::get<0>(*this); }
+  [[nodiscard]] const Value& GetValue() const&& { return std::get<0>(*this); }
+  [[nodiscard]] Value GetValue() && { return std::get<0>(std::move(*this)); }
+
+  [[nodiscard]] const NotFound& GetNotFound() const& { return std::get<1>(*this); }
+  [[nodiscard]] const NotFound& GetNotFound() const&& { return std::get<1>(*this); }
```

Comment:
Same question about the return type. I think it should match such of std::get.

Reply:
Done.